src/darwin/Framework/CHIP/MTRDeviceController.h

```diff
@@ -157,6 +158,21 @@ typedef void (^MTRDeviceConnectionCallback)(MTRBaseDevice * _Nullable device, NS
  */
 - (void)shutdown;
 
+/**
+ * Discover commissionable nodes.
+ */
+- (void)discoverCommissionableNodes;
+
+/**
+ * Set the Delegate for the device discovery
+ *
+ * @param[in] delegate The delegate the discovery process should use
+ *
+ * @param[in] queue The queue on which the callbacks will be delivered
+ */
+- (void)setDeviceDiscoveryDelegate:(id<MTRDeviceDiscoveryDelegate>)delegate
```

Comment:
Why do we have a separate delegate setter here, instead of having the "discover things" function take a block to call?

Reply:
I removed all discovery and ethernet pairing changes because they are useful for test but not a mandatory feature for Matter certification.